smoke_test.sh

```diff
@@ -103,14 +77,16 @@ PARGS=()
 CARGS=()
 if [ "${KCAT}" ]; then
     CMD=${KCAT}
+    PARGS=()
+    CARGS=()
 else
     CMD=docker
     PARGS=(run --rm --mount type=bind,source=${MSG_FILE},target=${MSG_FILE} --network=host edenhill/kcat:1.7.0)
     CARGS=(run --rm --network=host edenhill/kcat:1.7.0)
 fi
 
 ${CMD} \
- ${PARGS[@]} \
+ ${PARGS[@]+"${PARGS[@]}"} \
```

Comment:
I'm not familiar with this syntax and couldn't find it in the Bash manual. What's it doing?

Reply:
The background is here:
https://stackoverflow.com/questions/7577052/bash-empty-array-expansion-with-set-u

We could just require Bash 4.4 or higher, but that would mean Mac users having to install a new version of Bash (Mac ship with Bash 3.2 for licensing reasons).  

Since we already require a plethora of other packages installed, add `bash` wouldn't hurt.  We'd also need to adopt the `/usr/bin/env bash` shebang across all the scripts, so that the alternative bash is found from the PATH rather than `/bin/bash` which would find default on provided by the OS.

Reply:
> Since we already require a plethora of other packages installed, add `bash` wouldn't hurt. We'd also need to adopt the `/usr/bin/env bash` shebang across all the scripts, so that the alternative bash is found from the PATH rather than `/bin/bash` which would find default on provided by the OS.

This sounds reasonable and is a bit more flexible as well.

Reply:
I'll bring it up on list.